subcommand/flags/flag_duration_value.go

```diff
@@ -0,0 +1,64 @@
+package flags
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// Taken from https://github.com/hashicorp/consul/blob/master/command/flags/flag_duration_value.go
+// This was done so we don't depend on internal Consul implementation.
+
+// DurationValue provides a flag value that's aware if it has been set.
+type DurationValue struct {
```

Comment:
This type is currently only used by the [sync-catalog command](https://github.com/hashicorp/consul-k8s/blob/177d59b5d92f7cab73c67087072172d33f21dd3a/subcommand/sync-catalog/command.go#L46). I've also found that you could just use `time.Duration` and the flags would parse correctly (see [this](https://github.com/hashicorp/consul-k8s/blob/177d59b5d92f7cab73c67087072172d33f21dd3a/subcommand/lifecycle-sidecar/command.go#L26) example in the lifecycle-sidecar command). I'm wondering if we can replace the type of the `flagConsulWritePeriod` with `time.Duration`, then maybe we don't need this at all?

I have tried it out quickly, and it seems to be working ([branch](https://github.com/hashicorp/consul-k8s/tree/remove-flags-durationvalue)). This I think would also be fine to refactor afterward as it seems like it's expanding the scope of this task.

Reply:
I think it's fine to include, thanks for doing the leg work. Done.